packages/cli/src/metadata.ts

```diff
@@ -3,13 +3,36 @@ import type { JsonRpcProvider } from "@polkadot-api/json-rpc-provider"
 import * as fs from "node:fs/promises"
 import { V14, V15, metadata, v15 } from "@polkadot-api/substrate-bindings"
 import { WebSocketProvider } from "@polkadot-api/ws-provider/node"
-import { PROVIDER_WORKER_CODE } from "./smolldot-worker"
 import { Worker } from "node:worker_threads"
 import { getObservableClient } from "@polkadot-api/observable-client"
 import { filter, firstValueFrom } from "rxjs"
 import { EntryConfig } from "./papiConfig"
 import { dirname } from "path"
-import { WellKnownChain } from "./well-known-chains"
+import { fileURLToPath } from "url"
+import * as knownChains from "@polkadot-api/known-chains"
+import type {
+  MetadataWithRaw,
+  WorkerRequestMessage,
+  WorkerResponseMessage,
+} from "./metadataWorker"
+
+const workerPath = fileURLToPath(import.meta.resolve("./metadataWorker.js"))
```

Comment:
Keep in mind that support for `import.meta.resolve` was added to Node at `v20.6.0`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta/resolve#browser_compatibility
It's a rather recent version, I'd use `__dirname` instead since this only runs in Node.

Reply:
That's a good point - Originally I was using `import.meta.resolve` because the worker was in another node_modules package (previous versions of node would be `require.resolve`).

I think that Workers actually support relative paths, so if that works I might go with that.

Reply:
I forgot the relative path is always relative to the cwd 🤦 

`__dirname` doesn't exist in ESM, but it can be derived from `import.meta.url`, which was available in Node 10. I'll do that.

Reply:
> I forgot the relative path is always relative to the cwd 🤦
> 
> `__dirname` doesn't exist in ESM, but it can be derived from `import.meta.url`, which was available in Node 10. I'll do that.

Please don't. There is no reason why a brand new library should be supporting a "legacy" version of node. As of today node 18 is on maintenance, node 20 is "active"and node 22 is already "current".

There is no need for us to support node 18 at this point. We should just document the supported versions and be done with it.

Reply:
Also, FWIW `import.meta.resolve` works just fine with node 18.

What was released in version `v20.6.0` was the ability of using the second (and optional) parameter without the experimental flag. However, we are not using it, so we should be fine.